packages/edit-site/src/components/block-editor/index.js

```diff
@@ -51,28 +58,252 @@ const LAYOUT = {
 	alignments: [],
 };
 
+const FOCUSABLE_ENTITIES = [ 'wp_template_part', 'wp_navigation' ];
+
 export default function BlockEditor() {
-	const { setIsInserterOpened } = useDispatch( editSiteStore );
-	const { storedSettings, templateType, canvasMode, hasPageContentFocus } =
-		useSelect(
-			( select ) => {
-				const {
-					getSettings,
-					getEditedPostType,
-					getCanvasMode,
-					hasPageContentFocus: _hasPageContentFocus,
-				} = unlock( select( editSiteStore ) );
-
-				return {
-					storedSettings: getSettings( setIsInserterOpened ),
-					templateType: getEditedPostType(),
-					canvasMode: getCanvasMode(),
-					hasPageContentFocus: _hasPageContentFocus(),
-				};
-			},
-			[ setIsInserterOpened ]
+	const { templateType, hasPageContentFocus } = useSelect( ( select ) => {
+		const { getEditedPostType, hasPageContentFocus: _hasPageContentFocus } =
+			unlock( select( editSiteStore ) );
+
+		return {
+			templateType: getEditedPostType(),
+			hasPageContentFocus: _hasPageContentFocus(),
+		};
+	}, [] );
+
+	const settings = useSiteEditorSettings( templateType );
+
+	const BlockEditorComponent = getBlockEditorComponent( templateType );
+
+	return (
+		<BlockEditorComponent
+			settings={ settings }
+			templateType={ templateType }
+		>
+			{ hasPageContentFocus && <DisableNonPageContentBlocks /> }
+			<TemplatePartConverter />
+			<SidebarInspectorFill>
+				<BlockInspector />
+			</SidebarInspectorFill>
+
+			<SiteEditorCanvas
+				settings={ settings }
+				templateType={ templateType }
+			/>
+
+			<ReusableBlocksMenuItems />
+		</BlockEditorComponent>
+	);
+}
+
+/**
+ * Factory to isolate choosing the appropriate block editor
+ * component to handle a given template type/entity.
+ *
+ * @param {string} templateType the template (entity) type being edited
+ * @return {JSX.Element} the block editor component to use.
+ */
+function getBlockEditorComponent( templateType ) {
+	let Component = null;
+
+	switch ( templateType ) {
+		case 'wp_navigation':
+			Component = NavigationBlockEditor;
+			break;
+		default:
+			Component = DefaultBlockEditor;
+			break;
+	}
```

Comment:
Here is where we determine which editor provider should be handling which entity.